### PR TITLE
feat: open a modal to start a new message

### LINKED ISSUE #868: Select user for new DM from modal

### PR DESCRIPTION
**Pull request by elisealix22**
## What does this PR do?
This PR makes it so the new message button opens a modal to find someone new to message. You can either select one of the people you follow OR search for anyone on Lens and kick off a conversation.

Resolves #868

https://user-images.githubusercontent.com/556051/197285860-21c5a30f-d5d1-4946-b4ba-6693708eedc4.mov

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Search from the main nav bar and ensure it works
- [x] Search from new message and ensure it works
- [x] Ensure your followers show up in the modal
- [x] Ensure an empty state shows up if you do not follow anyone
- [x] Ensure selecting someone from the new message modal opens a new message component

### CODE REVIEW

**Comment by bigint** [comment]
Amazing one @elisealix22 can you please resolve the conflicts 🙏🏼

File: src/components/Messages/PreviewList.tsx [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+import Following from '@components/Profile/Following';
+import Search from '@components/Shared/Navbar/Search';
 import { Card } from '@components/UI/Card';
 import { GridItemFour } from '@components/UI/GridLayout';
+import { Modal } from '@components/UI/Modal';
 import { PageLoading } from '@components/UI/PageLoading';
 import useMessagePreviews from '@components/utils/hooks/useMessagePreviews';
+import type { Profile } from '@generated/types';
 import { PlusCircleIcon } from '@heroicons/react/outline';
+import { MailIcon } from '@heroicons/react/solid';
```
Comment (bigint) [inline src/components/Messages/PreviewList.tsx:11]: **suggestion:** outline icon is better

**Comment by bhavya2611** [comment]
LGTM

**Review by bhavya2611 — APPROVED** [review]

File: src/components/Messages/PreviewList.tsx (hunk 1 above)
Comment (elisealix22) [reply src/components/Messages/PreviewList.tsx]: good catch! updated.

**Comment by elisealix22** [comment]
Conflicts resolved! Thanks y'all 🙏🏻

**Pull request merged**